.github/workflows/run-nala-default.yml

```diff
@@ -1,25 +1,41 @@
-name: Run Nala Tests
+name: Nala E2E UI Tests
 
 on:
   push:
-    branches:
-      - stage
-      - main
+    branches: [stage, main]
   pull_request:
-    branches:
-      - stage
-      - main
+    branches: [stage, main]
     types: [opened, synchronize, reopened]
-  
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
 jobs:
   run-nala-tests:
-    name: Running Nala E2E UI Tests
+    name: ${{ matrix.project == 'mas' && 'Running MAS (all browsers)' || format('Running on • {0} browser', matrix.browser) }}
```

Comment:
M@S feature is going to move away to `adobecom/mas` repository soon, I think it is not worth investing time to particularly optimize it.